pymeasure/instruments/hcp/tc038.py

```diff
@@ -0,0 +1,128 @@
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pyvisa.constants import Parity
+
+
+class TC038(Instrument):
+    """
+    Communication with the HCP TC038 oven.
+
+    This is the older version with an AC power supply and AC heater.
+
+    It has parity or framing errors from time to time. Handle them in your
+    application.
+    """
+
+    def __init__(self, resourceName, address=1, timeout=1000,
+                 includeSCPI=False):
+        """
+        Initialize the communication.
+
+        Parameters
+        ----------
+        resourceName : str
+            name COM-Port.
+        address : int
+            address of the device. Should be between 1 and 99.
+        timeout : int
+            Timeout in ms.
+        """
+        super().__init__(resourceName, "TC038", timeout=timeout,
+                         write_termination="\r", read_termination="\r",
+                         parity=Parity.even,)
+        self.address = address
+
+        self.monitorTemperature()  # start to monitor the temperature
+
+    def write(self, command):
+        """
+        Send "command" to the oven with "commandData".
+
+        Parameters
+        ----------
+        command : string, optional
+            Command to be sent. Three chars indicating the type, and data for
+            the command, if necessary.
+        """
+        super().write(chr(2) + f"{self.address:02}" + "010"
+                      + command + chr(3))
+        # 010 is CPU (01) and time to wait (0), which are fix
+
+    def ask(self, command):
+        """
+        Send a command to the oven and read its response.
+
+        Parameters
+        ----------
+        command : string, optional
+            Command to be sent. Three chars indicating the type, and data for
+            the command, if necessary.
+
+        Returns
+        -------
+        string
+            response of the system.
+
+        """
+        return super().ask(chr(2) + f"{self.address:02}" + "010"
+                           + command + chr(3))
+
+    def monitor_temperature(self):
+        """Configure the oven to monitor the current temperature."""
+        self.ask(command="WRS" + "01" + "D0002")
+        # WRS in order to setup to monitor a word
+        # monitor 1 word
+        # monitor the word in register D0002
```

Comment:
A suggestion how this could be more generic (if in fact you can monitor other things than temperatures). The name reflects the association to `monitored_value`.

```suggestion
    def set_monitored_quantity(self, quantity='temperature'):
        """Configure the oven to monitor a certain register, by default the current temperature.        
        
        Valid entries are "temperature",...
        """
        monitored_registers: {'temperature': 'D002', 'something_else': 'D123',...}
        self.ask(command="WRS" + "01" + monitored_registers[quantity])
        # WRS in order to setup to monitor a word
        # monitor 1 word
        # monitor the word in the desired register
```

Reply:
You could monitor any value of the temperature controller, but I did not find anything useful to monitor besides the temperature.
Accessing the monitored value is a little faster and easier than accessing the values directly (you just request the monitored value, which is a single command, instead of requesting a specific value which requires a longer message). I do not see the benefit of reading the setpoint, or other internal configuration, with a fast readout.

Nevertheless, I edited it according to your suggestions, with a dictionary which could be expanded containing more addresses.